packages/server/src/utils/buildChatflow.ts

```diff
@@ -333,29 +343,80 @@ export const utilBuildChatflow = async (req: Request, socketIO?: Server, isInter
         const nodeModule = await import(nodeInstanceFilePath)
         const nodeInstance = new nodeModule.nodeClass({ sessionId })
 
-        let result = isStreamValid
-            ? await nodeInstance.run(nodeToExecuteData, incomingInput.question, {
-                  chatId,
-                  chatflowid,
-                  logger,
-                  appDataSource: appServer.AppDataSource,
-                  databaseEntities,
-                  analytic: chatflow.analytic,
-                  uploads: incomingInput.uploads,
-                  socketIO,
-                  socketIOClientId: incomingInput.socketIOClientId,
-                  prependMessages
-              })
-            : await nodeInstance.run(nodeToExecuteData, incomingInput.question, {
-                  chatId,
-                  chatflowid,
-                  logger,
-                  appDataSource: appServer.AppDataSource,
-                  databaseEntities,
-                  analytic: chatflow.analytic,
-                  uploads: incomingInput.uploads,
-                  prependMessages
-              })
+        const runParams = {
```

Comment:
i think the correct placement is [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/src/handler.ts#L357)

Reply:
Hi @HenryHengZJ are there events for "on user message" and "on assistant response"? Couldn't find it in the callback handlers space, which seems to be more related with Langchain events (as this PR is for tracking the high level user conversation)

Reply:
We don't need `on user message` as we can already have `input` from the callbacks in [AnalyticHandler](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/src/handler.ts#L317)

For instance, open a thread when `onChainStart`, and close when `onChainEnd`. I'd like to keep 3rd party integration on `components` level and try to reuse the callbacks that are already exposed